sourcegraph/about
# Guides: Optimizing Developer Onboarding

This closes #5206. It creates a template for guides and creates a new page for the developer onboarding use case. 

### Changelog

- Adds `GuideLayout` component
- Adds `optimizing-developer-onboarding.tsx`

### Notes

- ~~[Figma](https://www.figma.com/file/EVthdgOQ059ZS8S4Ixx91v/Optimizing-dev-onboarding?node-id=0%3A1)~~ and [meta](https://docs.google.com/spreadsheets/d/1VqW_U3RT1Vx-t7PksgwOadXsOaUTwGYvUnqRnN-fmb4/edit#gid=1572625403)
  - The Figma was updated on 5/23/22: [link](https://www.figma.com/file/VZeiowTUC4LviTGfN53r4g/Gated-content-landing-page-templates?node-id=0%3A1) 
- The issue proposes a new template for developing guide-related pages. However, their design is almost the same as the webinar template developed in #5344! I created a new component instead of reusing `WebinarLayout` because guides and webinars are different use cases, however, I am happy to refactor under the same component if we prefer to reduce technical debt. 
  - This would just mean renaming `WebinarLayout`, making `speaker` an optional field, and adjusting type in the header specified by the Figma.

### Test

1. Navigate to `/guides/optimizing-developer-onboarding`
2. Please check that layout matches Figma specs
3. Please check that a PDF opens properly when form is submitted
4. Ensure linting passes.
5. Ensure prettier has standardized the proposed changes.
6. Ensure dev and production builds work.
